src/cloudcasting/validation.py

```diff
@@ -104,12 +104,14 @@ def score_model_on_all_metrics(
     metrics: dict[str, list[TimeArray]] = {k: [] for k in metric_funcs}
 
     # we probably want to accumulate metrics here instead of taking the mean of means!
-    for num_batches_ran, (X, y) in enumerate(valid_dataloader):
+    loop_steps = len(valid_dataloader) if batch_limit is None else batch_limit
+
+    for i, (X, y) in tqdm(enumerate(valid_dataloader), total=loop_steps):
         y_hat = model(X)
         for metric_name, metric_func in metric_funcs.items():
             metrics[metric_name].append(metric_func(y_hat, y))
 
-        if num_termination_batches is not None and num_batches_ran >= num_termination_batches:
+        if batch_limit is not None and i >= batch_limit:
```

Comment:
Am I right in thinking you shouldn't need this now, given the loop should stop after loop_steps?

Reply:
`tqdm` doesn't actually stop the iteration at `loop_steps`. That just sets the length for of steps expected so it can be reflected in the progressbar. So it would keep looping without this